.github/workflows/esrupgrade.yml

```diff
@@ -9,144 +9,25 @@ on:
-    steps:
-      - name: Checkout mattermost-server
-        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-      - name: Run docker compose
-        run: |
-          cd build
-          docker-compose --no-ansi run --rm start_dependencies
-          cat ../tests/test-data.ldif | docker-compose --no-ansi exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
-          docker-compose --no-ansi exec -T minio sh -c 'mkdir -p /data/mattermost-test';
-          docker-compose --no-ansi ps
-      - name: Wait for docker compose
-        run: |
-          until docker network inspect ghactions_mm-test; do echo "Waiting for Docker Compose Network..."; sleep 1; done;
-          docker run --net ghactions_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://mysql:3306; do echo waiting for mysql; sleep 5; done;"
-          docker run --net ghactions_mm-test appropriate/curl:latest sh -c "until curl --max-time 5 --output - http://elasticsearch:9200; do echo waiting for elasticsearch; sleep 5; done;"
-      - name: Initialize the database with the source DB dump
-        run: |
-          curl $DB_DUMP_URL | zcat | docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS
-      - name: Run migration through script
-        run : |
-          cd scripts/esrupgrades
-          docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS < esr.5.37-7.8.mysql.up.sql
-      - name: Cleanup DB
-        run : |
-          cd scripts/esrupgrades
-          docker exec -i ghactions_mysql_1 mysql -AN $MYSQL_CONN_ARGS < esr.5.37-7.8.mysql.cleanup.sql
-      - name: Dump upgraded database
-        run: |
-          docker exec -i ghactions_mysql_1 mysqldump --skip-opt --set-gtid-purged=OFF $MYSQL_CONN_ARGS > dump.7.8.script.sql
-      - name: Cleanup dump and compress
-        run: |
-          # We skip the very last line, which simply contains the date of the dump
-          head -n -1 dump.7.8.script.sql | gzip > dump.7.8.script.sql.gz
-      - name: Upload dump
-        uses: actions/upload-artifact@v3
-        with:
-          name: upgraded-dump-script
-          path: dump.7.8.script.sql.gz
-  esr-upgrade-diff:
-    runs-on: ubuntu-latest-8-cores
-    needs:
-      - esr-upgrade-server
-      - esr-upgrade-script
-    steps:
-      - name: Retrieve dumps
-        uses: actions/download-artifact@v3
-      - name: Diff dumps
-        run: |
-          gzip -d upgraded-dump-server/dump.7.8.server.sql.gz
-          gzip -d upgraded-dump-script/dump.7.8.script.sql.gz
-          diff upgraded-dump-server/dump.7.8.server.sql upgraded-dump-script/dump.7.8.script.sql > diff.7.8.sql
-      - name: Upload diff
-        if: failure() # Upload the diff only if the previous step failed; i.e., if the diff is non-empty
-        uses: actions/upload-artifact@v3
-        with:
-          name: dumps-diff
-          path: diff.7.8.sql
+  esr-upgrade-5_37-7_8:
```

Comment:
Thinking out loud, but given the overall complexity, I wonder if we should add a 'should-fail' job (and maybe an 'expect-failure' parameter)? The idea being to explicitly test the script itself by providing inputs that are expected to trigger a diff, and just "pass" that. Thoughts?

Reply:
There are numerous fail scenarios but did you had something specific in your mind? I'm thinking cases we explicitly don't touch. Eg. collation issues etc. Is it something like that?

Reply:
I'm probably just being paranoid, but given all the moving parts, I was just worried that maybe a regression might result in the tests doing nothing, but they would still pass under current semantics.